cspatients/util.py

```diff
@@ -87,42 +79,49 @@ def save_model_changes(data):
         200 if changes were made. 400 if the PatientEntry could not be found.
 
     """
-    all_values = get_patient_dict(data).update(get_patiententry_dict(data))
+
+    changes_dict = get_rp_dict(data, "entrychanges")
     try:
-        current_patiententry = PatientEntry.objects.get(
-            patient_id=all_values['patient_id']
+        patiententry = PatientEntry.objects.get(
+            patient_id=changes_dict['patient_id']
+            )
+        del changes_dict['patient_id']
+        for label in changes_dict:
+            if label in PATIENTENTRY_FIELDS:
+                patiententry.__dict__[label] = changes_dict[label]
+            if label in PATIENT_FIELDS:
+                patiententry.__dict__[label] = changes_dict[label]
+        return 200
+    except PatientEntry.DoesNotExist:
+        return 400
+    except Patient.DoesNotExist:
+        return 400
+
+
+def save_model(post_data):
```

Comment:
This method makes me feel uncomfortable. I'm sure either the Serializer or a form should handle this for you.